crates/agent-sql/src/publications.rs

```diff
@@ -88,14 +88,57 @@ pub async fn update_live_specs(
                 $14::text[]
             )
         ),
-        joined(catalog_name, spec_type, spec, built_spec, expect_build_id, reads_from, writes_to, image, image_tag, data_plane_id, is_touch, dependency_hash, last_build_id, next_pub_id) as (
+        joined(catalog_name, spec_type, spec, built_spec, expect_build_id, reads_from, writes_to, image, image_tag, data_plane_id, is_touch, dependency_hash, last_build_id, next_pub_id, controller_task_id) as (
             select
-                inputs.*,
+                inputs.catalog_name,
+                inputs.spec_type,
+                inputs.spec,
+                inputs.built_spec,
+                inputs.expect_pub_id,
+                inputs.reads_from,
+                inputs.writes_to,
+                inputs.image,
+                inputs.image_tag,
+                inputs.data_plane_id,
+                inputs.is_touch,
+                inputs.dependency_hash,
                 case when ls.spec is null then '00:00:00:00:00:00:00:00'::flowid else ls.last_build_id end as last_build_id,
-                case when inputs.is_touch then ls.last_pub_id else $1::flowid end as next_pub_id
+                case when inputs.is_touch then ls.last_pub_id else $1::flowid end as next_pub_id,
+                case when ls.controller_task_id is null then internal.id_generator() else ls.controller_task_id end as controller_task_id
             from inputs
             left outer join live_specs ls on ls.catalog_name = inputs.catalog_name
         ),
+        create_controller_tasks as (
+            insert into internal.tasks (task_id, task_type, wake_at, inbox)
+            select
+                controller_task_id,
+                2,
+                now(),
+                array[json_build_array(
```

Comment:
This is replicating a lot of innards of `internal.send_to_task`.

Can you separate it into two CTE stages ? The first would create task rows which don't exist (on conflict do nothing), setting just the task type, task ID, and parent ID as zero.

The second CTE would then call `internal.send_to_task` across all now-existing task rows.

Reply:
Yeah, I should have commented this. The reason I did this was because we frequently update hundreds of rows at a time, and I was concerned about the performance of repeatedly calling `send_to_task`, which IIUC would fetch each row separately vs doing a join. TBH I'd be quite willing to call that a 'premature optimization' and just use `send_to_task`. It's hard to do meaningful profiling outside of prod, so I was just going off vibes here and I don't really have useful intuitions. But if there's a perf issue with repeatedly calling `send_to_task`, then I'd expect it to show up pretty quickly in prod.

Reply:
Fair. I would hope that it's not gonna be a problem, even if doing hundreds of updates, but I can't say that with any certainty. This is fine, I suppose, but it's a little eye watering.